SecondHomework/Logistic/Program.cs

```diff
@@ -59,64 +72,93 @@
     string? ID = Console.ReadLine();
     if (Guid.TryParse(ID, out Guid id))
     {
-        List<Parcel> parcel = parcelManager.GetParcels();
-        Result result = parcelManager.RemoveParcel(id, parcel);
+        List<Parcel> parcels = await parcelManager.GetParcelsAsync();
+        Result result = await parcelManager.RemoveParcelAsync(id);
         bool isRemoved = result.Success;
-        Console.WriteLine(result.Message);
+        logger.PrintMessage(result.Message ?? "No message provided.");
 
         if (isRemoved)
         {
-            fileManager.Save(parcel);
-        }
-        foreach (var parcels in parcelManager.GetParcels())
-        {
-            Console.WriteLine(parcels.ToString());
+            await fileManager.SaveParcelsAsync(parcels);
```

Comment:
You should not call it here, it is already being called inside ParcelManager

Reply:
i've found that =(